### PR TITLE
Use SimulatedBackend for TestDirectFund

### PR DESCRIPTION
**Pull request by kerzhner**
This PR incorporates the SimulatedBackendChainService (which is a thin wrapper around the EthChainService) into the TestDirectFund test.

The goal of the PR is to testdrive transaction submission and event listening in a more rich environment than the rudimentary `EthChainServiceTest`.

### CODE REVIEW

File: client/engine/chainservice/simulated_backend_chainservice_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+	out := cs.SubscribeToEvents(ethAccounts[0].From)
+	// Submit transactiom
+	cs.SendTransaction(testTx)
+
+	// Check that the recieved event matches the expected event
+	receivedEvent := <-out
+	expectedEvent := DepositedEvent{CommonEvent: CommonEvent{channelID: channelID, BlockNum: 2}, Holdings: testDeposit}
+	if diff := cmp.Diff(expectedEvent, receivedEvent, cmp.AllowUnexported(CommonEvent{})); diff != "" {
+		t.Fatalf("Clone: mismatch (-want +got):\n%s", diff)
```
Comment (geoknee) [inline client/engine/chainservice/simulated_backend_chainservice_test.go:40]:
```suggestion
		t.Fatalf("Received event did not match expectation; (-want +got):\n%s", diff)
```

File: client/engine/chainservice/simulated_backend_chainservice_test.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	}
+
+	// Not sure if this is necessary
```
Comment (geoknee) [inline client/engine/chainservice/simulated_backend_chainservice_test.go:43]:
```suggestion

```

**Review by geoknee — APPROVED** [review]
👏

File: client/engine/chainservice/automine_chainservice.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+// NewAutomineChainService constructs a chain service that submits transactions to a NitroAdjudicator
+// and listens to events from an eventSource
+func NewAutomineChainService(sim transactionProcessor, na *NitroAdjudicator.NitroAdjudicator, naAddress common.Address,
+	txSigner *bind.TransactOpts, es eventSource) *AutomineChainService {
+	return &AutomineChainService{sim: sim, EthChainService: NewEthChainService(na, naAddress, txSigner, es)}
+}
+
+// SendTransaction sends the transaction and blocks until it has been submitted.
+func (ecs *AutomineChainService) SendTransaction(tx protocols.ChainTransaction) {
```
[the remark this answers is not in the document]
Comment (kerzhner) [reply client/engine/chainservice/automine_chainservice.go]: Looks like this code is from an earlier commit. The file and struct were renamed in later commits. I should have cleaned up commit history.

File: client/engine/chainservice/simulated_backend_chainservice.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+)
+
+var ErrUnableToAssignBigInt = errors.New("simulated_backend_chainservice: unable to assign BigInt")
+
+type transactionProcessor interface {
+	Commit()
+}
+
+// SimulatedBackendChaneService extends EthChainService to automatically mine a block for every transaction
+type SimulatedBackendChaneService struct {
```
[the remark this answers is not in the document]
Comment (kerzhner) [reply client/engine/chainservice/simulated_backend_chainservice.go]: Good catch. Fixed in 937ce172

File: client_test/helpers_test.go [hunk 5]
```diff
@@ -66,7 +66,7 @@ func waitTimeForCompletedObjectiveIds(t *testing.T, client *client.Client, timeo
 }
 
 // setupClient is a helper function that contructs a client and returns the new client and its store.
-func setupClient(pk []byte, chain *chainservice.MockChain, msgBroker messageservice.Broker, logDestination io.Writer, meanMessageDelay time.Duration) (client.Client, store.Store) {
```
[the remark this answers is not in the document]
Comment (kerzhner) [reply client_test/helpers_test.go:69]: Yep, as long as the `MockChain` struct adheres to the ChainService interface. Are there other concerns that I am not seeing?

**Pull request merged**